TeachingRecordSystem/tests/TeachingRecordSystem.SupportUi.Tests/PageTests/SupportTasks/ApiTrnRequests/Resolve/MatchesTests.cs

```diff
@@ -452,13 +452,97 @@ public async Task Post_CreateNewRecordChosen_UpdatesStateAndRedirects()
         // Assert
         Assert.Equal(StatusCodes.Status302Found, (int)response.StatusCode);
         Assert.Equal(
-            $"/support-tasks/api-trn-requests/{supportTask.SupportTaskReference}/merge?{journeyInstance.GetUniqueIdQueryParameter()}",
+            $"/support-tasks/api-trn-requests/{supportTask.SupportTaskReference}/check-answers?{journeyInstance.GetUniqueIdQueryParameter()}",
             response.Headers.Location?.OriginalString);
 
         journeyInstance = await ReloadJourneyInstance(journeyInstance);
         Assert.Equal(personId, journeyInstance.State.PersonId);
     }
 
+    [Fact]
+    public async Task Post_PersonIdChangedFromState_ClearsPersonAttributes()
+    {
+        // Arrange
+        var applicationUser = await TestData.CreateApplicationUserAsync();
+        var supportTask = await TestData.CreateApiTrnRequestSupportTaskAsync(applicationUser.UserId);
+
+        var selectedPersonId = ResolveApiTrnRequestState.CreateNewRecordPersonIdSentinel;
+
+        var journeyInstance = await CreateJourneyInstance(
+            supportTask.SupportTaskReference,
+            new ResolveApiTrnRequestState
+            {
+                PersonId = supportTask.TrnRequestMetadata!.Matches!.MatchedRecords.First().PersonId,
+                FirstNameSource = ResolveApiTrnRequestState.PersonAttributeSource.ExistingRecord,
+                MiddleNameSource = ResolveApiTrnRequestState.PersonAttributeSource.ExistingRecord,
+                LastNameSource = ResolveApiTrnRequestState.PersonAttributeSource.ExistingRecord,
+                DateOfBirthSource = ResolveApiTrnRequestState.PersonAttributeSource.ExistingRecord,
+                EmailAddressSource = ResolveApiTrnRequestState.PersonAttributeSource.ExistingRecord,
+                NationalInsuranceNumberSource = ResolveApiTrnRequestState.PersonAttributeSource.ExistingRecord
+            });
+
+        var request = new HttpRequestMessage(
+            HttpMethod.Post,
+            $"/support-tasks/api-trn-requests/{supportTask.SupportTaskReference}/matches?{journeyInstance.GetUniqueIdQueryParameter()}")
+        {
+            Content = new FormUrlEncodedContentBuilder() { { "PersonId", selectedPersonId } }
+        };
+
+        // Act
+        var response = await HttpClient.SendAsync(request);
+
+        // Assert
+        journeyInstance = await ReloadJourneyInstance(journeyInstance);
+        Assert.Null(journeyInstance.State.FirstNameSource);
+        Assert.Null(journeyInstance.State.MiddleNameSource);
+        Assert.Null(journeyInstance.State.LastNameSource);
+        Assert.Null(journeyInstance.State.DateOfBirthSource);
+        Assert.Null(journeyInstance.State.EmailAddressSource);
+        Assert.Null(journeyInstance.State.NationalInsuranceNumberSource);
```

Comment:
I'm not quite sure I understand why these should be null now - tried to read the code but I'm not seeing it

Reply:
It's so that if a user goes from Matches to Merge then CheckAnswers (so the `*Source` properties are populated) then back to Matches and changes the Person then the `Source` attributes get reset, effectively forcing the user to go through Merge again.